src/registry.py

```diff
@@ -1,9 +1,83 @@
+class SoundEventUUIDInfo:
+    """
+    Tracks the UUIDs related to a specific sound event id.
+    """
+    def __init__(self, registry):
+        self.registry = registry
+        self._uuids = set()
+
+    def add_sound(self, sound):
+        """
+        Adds the sound uuid to the set of uuids (per sound event id).
+        """
+        self._uuids.add(sound.uuid)
+
+    def remove_sound(self, sound):
+        """
+        Removes the sound uuid from the set of uuids (per sound event id).
+        """
+        self._uuids.remove(sound.uuid)
+
+    @property
+    def uuids(self):
+        """
+        Gets the UUIDs for this sound event id info.
+        """
+        return self._uuids
+
+
+class SoundEventsRegistry:
+    """
+    Tracks information related too a given sound event id.
+    """
+    def __init__(self, registry):
+        self.registry = registry
+        self._sound_events = {}
+
+    def add_sound(self, sound):
+        """
+        Adds sound event id information related to a given sound.
+        """
+        if sound.sound_event_id not in self._sound_events:
+            self._sound_events[sound.sound_event_id] = \
+                SoundEventUUIDInfo(self.registry)
+        self._sound_events[sound.sound_event_id].add_sound(sound)
+
+    def remove_sound(self, sound):
+        """
+        Removes sound event id information related to a given sound.
+        """
+        self._sound_events[sound.sound_event_id].remove_sound(sound)
```

Comment:
This should check if `self._sound_events[sound.sound_event_id]` exists before, no?